packages/flutter_tools/lib/src/commands/update_packages.dart

```diff
@@ -131,9 +121,9 @@ class UpdatePackagesCommand extends FlutterCommand {
   Future<void> _downloadCoverageData() async {
     final String urlBase = globals.platform.environment['FLUTTER_STORAGE_BASE_URL'] ?? 'https://storage.googleapis.com';
     final Uri coverageUri = Uri.parse('$urlBase/flutter_infra_release/flutter/coverage/lcov.info');
-    final List<int> data = await _net.fetchUrl(coverageUri);
+    final List<int> data = (await _net.fetchUrl(coverageUri))!;
```

Comment:
Seems like the null should be handled if the download fails and this returns null.

Reply:
done